### PR TITLE
NSPathUtilities: Implement NSString.fileSystemRepresentation

### PR DESCRIPTION
**Pull request by spevans**
Fixes https://bugs.swift.org/browse/SR-10364

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test

File: Foundation/NSPathUtilities.swift [hunk 1]
```diff
@@ -537,8 +537,8 @@ extension NSString {
         return path + "/"
     }
     
-    public var fileSystemRepresentation : UnsafePointer<Int8> {
-        NSUnimplemented()
+    public var fileSystemRepresentation: UnsafePointer<Int8> {
+        return FileManager.default.fileSystemRepresentation(withPath: self._swiftObject)
```
Comment (compnerd) [inline Foundation/NSPathUtilities.swift]: Isn't this introducing a memory leak?  @millenomi

Comment (spevans) [reply Foundation/NSPathUtilities.swift]: Indeed it is, its the same as `FileManager.fileSystemRepresentation(withPath:)` and after use `.deallocate()` needs to be on Linux.

On macOS this is normally an autoreleasing pointer. Hopefully at some point a new method will be added which uses a closure.

Comment (compnerd) [reply Foundation/NSPathUtilities.swift]: Can we at least have some way to indicate that this introduces a memory leak?

Comment (millenomi) [reply Foundation/NSPathUtilities.swift]: While I'm okay taking a patch that brings this to parity with `.fSR(wP:)`, `.fSR(wP:)` is not at parity with Darwin, and I'll be looking at introducing a block-taking experimental version.

**Pull request merged**